src/miner.cpp

```diff
@@ -855,6 +1041,56 @@ void StakeMiner(CWallet *pwallet)
     RenameThread("grc-stake-miner");
 
     MinerAutoUnlockFeature(pwallet);
+    
+    // Parse StakeSplit and SideStaking flags.
+    bool fEnableStakeSplit = GetBoolArg("-enablestakesplit");
+    LogPrintf("StakeMiner: fEnableStakeSplit = %u", fEnableStakeSplit);
+
+    bool fEnableSideStaking = GetBoolArg("-enablesidestaking");
+    LogPrintf("StakeMiner: fEnableSideStaking = %u", fEnableSideStaking);
+
+    vector<string> vSubParam;
+    SideStakeAlloc vSideStakeAlloc;
+    int64_t nMinStakeSplitValue;
+    double dEfficiency;
+    
+    // If side staking is enabled, parse destinations and allocations.
+    if (fEnableSideStaking)
+    {
+        if (mapArgs.count("-sidestake") && mapMultiArgs["-sidestake"].size() > 0)
+        {
+            for (auto const& sSubParam : mapMultiArgs["-sidestake"])
+            {
+                ParseString(sSubParam, ',', vSubParam);
+                vSideStakeAlloc.push_back(std::pair<std::string, double>(vSubParam[0], atof(vSubParam[1].c_str()) / 100.0));
+                vSubParam.clear();
+            }
+        }
+
+        for(auto iter : vSideStakeAlloc)
```

Comment:
do we have to go through this loop? could the logging not be in the mapMultiArgs["-sidestake"] area before the push_back? (yes u prob would make a floating point var for ease of this). probably wouldn't be a significant impact anyway. if it stays there then const as no need to copy. also if there is no sidestake settings we would hit this also even thou it would pan out to nothing if im correct.

Correct me if i'm wrong ofcourse. but it is shaping up pretty nice for a feature

Reply:
Yes. I wanted to make sure stuff was making it into the vector < pair <>> properly. The loops can be combined. The whole thing is conditioned upon the side stake flag, although someone could put 1 for the flag and then include no allocations... but that is ok. It will hit the for and then move on...